src/Symfony/Component/Messenger/Worker.php

```diff
@@ -282,4 +326,79 @@ public function getMetadata(): WorkerMetadata
     {
         return $this->metadata;
     }
+
+    public function handleFutures(string $transportName, $parallelProcessLimit): void
+    {
+        $toHandle = self::$futures;
+        self::$futures = [];
+
+        if (!$toHandle) {
+            return;
+        }
+
+        $futuresReceived = [];
+        $envelopesAssociated = [];
+
+        foreach ($toHandle as $combo) {
+            $futuresReceived[] = $combo[0];
+            $envelopesAssociated[] = $combo[1];
+        }
+
+        [$errorsFromAwait, $executions] = awaitAll($futuresReceived);
+
+        foreach ($errorsFromAwait as $index => $error) {
+            try {
+                $execution = $futuresReceived[$index]->await();
+                $envelope = $execution->await();
```

Comment:
Since we already know an error was thrown on the previous line, isn't this unreachable code?